iothub/device/src/Authentication/IotHubConnectionInfo.cs

```diff
@@ -187,5 +181,54 @@ public Uri BuildLinkAddress(string path)
 
             return builder.Uri;
         }
+
+        public bool IsPooling()
+        {
+            return (AuthenticationModel != AuthenticationModel.X509) && (AmqpTransportSettings?.ConnectionPoolSettings?.Pooling ?? false);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is IotHubConnectionInfo iotHubConnectionInfo
+                && GetHashCode() == iotHubConnectionInfo.GetHashCode()
+                && Equals(DeviceId, iotHubConnectionInfo.DeviceId)
+                && Equals(HostName, iotHubConnectionInfo.HostName)
+                && Equals(ModuleId, iotHubConnectionInfo.ModuleId)
+                && Equals(AmqpTransportSettings.Protocol, iotHubConnectionInfo.AmqpTransportSettings.Protocol)
+                && Equals(AuthenticationModel.GetHashCode(), iotHubConnectionInfo.AuthenticationModel.GetHashCode());
+        }
+
+        public override int GetHashCode()
```

Comment:
The hashcode algorithm used only these fields for creating the hash (in DeviceIdentity), so I left them as-is.

Reply:
Probably worth a code comment then, otherwise it is unclear why only some properties are included?